### DISCUSSION TITLE
Refresh not supported and does not have appropriate auth credentials in kubeconfig

### DISCUSSION THREAD

**Discussion by ImYourBoss0808** [kubernetes-client/csharp#1410] [Q&A] [answered]
I use azure cli to create the kubeconfig file. when I use the client library for .net it shows an error saying refresh not supported.

I check for some suggestions and they said that I should use kubelogin, I tried the kubelogin convert but when  run my application, the error says that I do not have appropriate auth credentials. 

Is there a way to refresh the token using the client library for dotnet?since the kubelogin is not working.

**Comment by tg123** [comment, marked as answer]
this sdk does support kubelogin with token auto refresh

any error?

**Comment by ImYourBoss0808** [comment]
Hey, it's working now. 
But I have a question,
Do I need to keep the installation of kubelogin plugin after I convert the kubeconfig file?
Or can I uninstall that plugin already? thanks

**Comment by tg123** [comment]
`kubelogin` should be in PATH
sdk will invoke it to get token

**Reply by ImYourBoss0808** [reply]
One more question, if I create a kubeconfig file using a service account, can I still use the kubelogin script kubelogin convert-kubeconfig -l azurecli to convert the kubeconfig file?, currently I use my azure credential to create the kubeconfig file using the

**Reply by tg123** [reply]
`service account` or service principal

**Reply by ImYourBoss0808** [reply]
service account

**Reply by ImYourBoss0808** [reply]
or do you suggest to use a service principal account instead?
